js/renovation/ui/toolbar/toolbar.tsx

```diff
@@ -1,29 +1,51 @@
-/* eslint-disable max-classes-per-file */
-
 import {
-  Component, JSXComponent,
+  Component, Consumer, JSXComponent,
 } from '@devextreme-generator/declarations';
 /* eslint-disable import/named */
 import LegacyToolbar from '../../../ui/toolbar';
 
 import { DomComponentWrapper } from '../common/dom_component_wrapper';
 import { ToolbarProps } from './toolbar_props';
+import { isDefined, isString } from '../../../core/utils/type';
+import { ConfigContext, ConfigContextValue } from '../../common/config_context';
+import { resolveRtlEnabled } from '../../utils/resolve_rtl';
+
+export const viewFunction = ({ props, rtl, restAttributes }: Toolbar): JSX.Element => {
+  const { items } = props;
+  if (isDefined(items)) {
+    // eslint-disable-next-line @typescript-eslint/prefer-for-of
+    for (let index = 0; index < items.length; index += 1) {
+      const item = items[index];
+      if (!isString(item)) {
+        if (!isDefined(item.options)) {
+          item.options = {};
+        }
+        item.options.rtlEnabled = item.options.rtlEnabled ?? rtl;
+      }
+    }
+  }
 
-export const viewFunction = ({
-  props,
-  restAttributes,
-}: Toolbar): JSX.Element => (
-  <DomComponentWrapper
-    componentType={LegacyToolbar}
-    componentProps={props}
-  // eslint-disable-next-line react/jsx-props-no-spreading
-    {...restAttributes}
-  />
-);
+  return (
+    <DomComponentWrapper
+      componentType={LegacyToolbar}
+      componentProps={props}
+      // eslint-disable-next-line react/jsx-props-no-spreading
+      {...restAttributes}
+    />
+  );
+};
 
 @Component({
   defaultOptionRules: null,
   view: viewFunction,
 })
 
-export class Toolbar extends JSXComponent<ToolbarProps>() {}
+export class Toolbar extends JSXComponent<ToolbarProps>() {
+  @Consumer(ConfigContext)
+  config?: ConfigContextValue;
+
+  get rtl(): boolean {
```

Comment:
Shall we rename it to rtlEnabled?

Reply:
Y think, no.
It will have the same name as rtlEnabled false and may confuse

Reply:
Everywhere we use 'rtlEnabled' to specify whether rtl should be enabled or not.
This function calculates new 'rtlEnabled' from several sources.
It's better to keep 'rtlEnabled' in its name and add a prefix/suffix that easily describes "what is the difference between props.rtlEnabled and this property"